runtime/list.go

```diff
@@ -170,6 +170,38 @@ func listCount(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
 	return seqCount(f, args[0], args[1])
 }
 
+func listRemove(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
+	if raised := checkMethodArgs(f, "remove", args, ListType, ObjectType); raised != nil {
+		return nil, raised
+	}
+	v := args[1]
+	i := -1
+	pred := func(o *Object) (bool, *BaseException) {
+		i++
+		eq, raised := Eq(f, o, v)
+		if raised != nil {
+			return false, raised
+		}
+		ret, raised := IsTrue(f, eq)
+		if raised != nil {
+			return false, raised
+		}
+		return ret, nil
+	}
+	l := toListUnsafe(args[0])
+	l.mutex.RLock()
+	defer l.mutex.RUnlock()
```

Comment:
Avoid defer except in cases where performance doesn't matter. There is a huge tax to using defer. It may help to structure your code like this to avoid doing multiple unlocks:

```
foundEqItem, raised := seqFindFirst(f, args[0], pred)
if raised == nil {
  if foundEqItem {
    raised = f.RaiseType(ValueErrorType, "list.remove(x): x not in list")
  } else {
    l.elems = append(l.elems[:i], l.elems[i+1:]...)
  }
}
l.mutex.Unlock()
if raised != nil {
  return nil, raised
}
return None, nil
```